[CECO-884] Add sidecar injection basic config in operator (#1190)
DataDog/datadog-operator: docs/configuration.v2alpha1.md

| Parameter | Description |
| --------- | ----------- |
| features.admissionController.agentCommunicationMode | AgentCommunicationMode corresponds to the mode used by the Datadog application libraries to communicate with the Agent. It can be "hostip", "service", or "socket". |
| features.admissionController.agentSidecarInjection.clusterAgentCommunicationEnabled | ClusterAgentCommunicationEnabled enables communication between Agent sidecars and the Cluster Agent. Default : true |

> | features.admissionController.agentSidecarInjection.enabled | Enabled enables Sidecar injections. Default: false |

Reviewer: ```suggestion
| features.admissionController.agentSidecarInjection.enabled | Enables Sidecar injections. Default: false |
```

| Parameter | Description |
| --------- | ----------- |
| features.admissionController.agentCommunicationMode | AgentCommunicationMode corresponds to the mode used by the Datadog application libraries to communicate with the Agent. It can be "hostip", "service", or "socket". |

> | features.admissionController.agentSidecarInjection.clusterAgentCommunicationEnabled | ClusterAgentCommunicationEnabled enables communication between Agent sidecars and the Cluster Agent. default : true |

Reviewer: ```suggestion
| features.admissionController.agentSidecarInjection.clusterAgentCommunicationEnabled | ClusterAgentCommunicationEnabled enables communication between Agent sidecars and the Cluster Agent. Default : true |
```

> | features.admissionController.agentSidecarInjection.enabled | Enabled enables Sidecar injections. Default: true |

Reviewer: ```suggestion
| features.admissionController.agentSidecarInjection.enabled | Enables Sidecar injections. Default: true |
```

> | features.admissionController.agentSidecarInjection.imageName | ImageName overrides the default agent image name for the Agent sidecar. |

Reviewer: ```suggestion
| features.admissionController.agentSidecarInjection.imageName | ImageName overrides the default Agent image name for the Agent sidecar. |
```

> | features.admissionController.agentSidecarInjection.imageTag | ImageTag overrides the default agent image tag for the Agent sidecar. |

Reviewer: ```suggestion
| features.admissionController.agentSidecarInjection.imageTag | ImageTag overrides the default Agent image tag for the Agent sidecar. |
```

> | features.admissionController.agentSidecarInjection.provider | Provider defines the Currently only "fargate" is supported If not specified, It will automatically set to "fargate" To use the feature in other environments (including local testing) omit the config. ref: https://docs.datadoghq.com/integrations/eks_fargate |

Reviewer: ```suggestion
| features.admissionController.agentSidecarInjection.provider | Provider defines the provider for the Datadog Agent sidecar injection. Currently, only "fargate" is supported. If not specified, it is automatically be set to "fargate". To use the feature in other environments (including local testing) omit the config. See also: https://docs.datadoghq.com/integrations/eks_fargate |
```
Please let me know if this is accurate. It seems like some sentences were merged together, so it wasn't clear to me.

| features.admissionController.agentSidecarInjection.image.tag | Define the image tag to use. To be used if the Name field does not correspond to a full image string. |
| features.admissionController.agentSidecarInjection.provider | Provider is used to add infrastructure provider-specific configurations to the Agent sidecar. Currently only "fargate" is supported. To use the feature in other environments (including local testing) omit the config. See also: https://docs.datadoghq.com/integrations/eks_fargate |
| features.admissionController.agentSidecarInjection.registry | Registry overrides the default registry for the sidecar Agent. |

> | features.admissionController.cwsInstrumentation.enabled | Enable the CWS Instrumentation admission controller endpoint Default: false |

Reviewer: ```suggestion
| features.admissionController.cwsInstrumentation.enabled | Enable the CWS Instrumentation admission controller endpoint. Default: false |
```

Reviewer 2: +1
